ext/pcntl/pcntl.c

```diff
@@ -140,9 +140,17 @@ PHP_MSHUTDOWN_FUNCTION(pcntl)
 PHP_RSHUTDOWN_FUNCTION(pcntl)
 {
 	struct php_pcntl_pending_signal *sig;
+	int signo;
+	zval *handle;
+
+	/* Remove all registered signal handlers */
+	for (signo = 1; signo < PCNTL_G(num_signals); ++signo) {
+		handle = zend_hash_index_find(&PCNTL_G(php_signal_table), signo);
```

Comment:
We could iterate on `php_signal_table` with `ZEND_HASH_FOREACH_NUM_KEY_VAL` to avoid `num_signals` lookups

Reply:
Excellent! I was trying to find such method myself, but couldn't. Thanks!